### ISSUE TITLE
Breaking changes in docker

### ISSUE THREAD

**Issue by saartochner** [spulec/moto#3359]
## Reporting Bugs

Hey, 
If there is no docker deamon running, then using `mock_lambda` with the newest version of docker-py (4.3.1) yields the following error:

```
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', ConnectionRefusedError(61, 'Connection refused'))
```
Version 4.3.0 does not have this problem.

**Comment by bblommers** [comment]
Thanks for raising this @saartochner. Looks like it's related to https://github.com/docker/docker-py/issues/2665 and https://github.com/docker/docker-py/issues/2676

**Comment by bblommers** [comment]
Little late, but this should now be fixed as of moto >= 1.3.16.dev110
